src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.cs

```diff
@@ -198,22 +198,29 @@ protected virtual void SetupNewTextView(IVsTextView textView)
                 new StandaloneCommandFilter(
                     v, Package.ComponentModel).AttachToVsTextView());
 
+            var isMetadataAsSource = false;
+            var collapseAllImplementations = false;
+
             var openDocument = wpfTextView.TextBuffer.AsTextContainer().GetRelatedDocuments().FirstOrDefault();
+            if (openDocument?.Project.Solution.Workspace is MetadataAsSourceWorkspace masWorkspace)
+            {
+                isMetadataAsSource = true;
 
-            // If the file is metadata as source, and the user has the preference set to collapse them, then
-            // always collapse all metadata as source
-            var globalOptions = this.Package.ComponentModel.GetService<IGlobalOptionService>();
-            var masWorkspace = openDocument?.Project.Solution.Workspace as MetadataAsSourceWorkspace;
-            var options = BlockStructureOptionsStorage.GetBlockStructureOptions(globalOptions, openDocument.Project.Language, isMetadataAsSource: masWorkspace is not null);
```

Comment:
This was the bad line of code. `openDocument` can be null, so `openDocument.Project` threw an NRE. Moving it inside the if fixes the issue.